src/panels/lovelace/components/hui-input-list-editor.ts

```diff
@@ -0,0 +1,100 @@
+import {
+  html,
+  LitElement,
+  PropertyDeclarations,
+  TemplateResult,
+} from "lit-element";
+import "@polymer/paper-input/paper-input";
+
+import { HomeAssistant } from "../../../types";
+import { fireEvent } from "../../../common/dom/fire_event";
+
+import { EditorTarget } from "../editor/types";
+
+export class HuiInputListEditor extends LitElement {
+  protected value?: string[];
+  protected hass?: HomeAssistant;
+  protected heading?: string;
+  protected inputLabel?: string;
+
+  static get properties(): PropertyDeclarations {
+    return {
+      hass: {},
+      value: {},
+      heading: {},
+      inputLabel: {},
+    };
+  }
+
+  protected render(): TemplateResult | void {
+    if (!this.value) {
+      return html``;
+    }
+
+    return html`
+      ${this.renderStyle()}
+      <h3>${this.heading}</h3>
+      <div class="entries">
+        ${this.value.map((listEntry, index) => {
+          return html`
+            <paper-input
+              label="${this.inputLabel}"
+              .value="${listEntry}"
+              .configValue="${"entry"}"
+              .index="${index}"
+              @value-changed="${this._valueChanged}"
+            ></paper-input>
+          `;
+        })}
+        <paper-input
+          label="${this.inputLabel}"
+          @change="${this._addEntity}"
+        ></paper-input>
+      </div>
+    `;
+  }
+
+  private _addEntity(ev: Event): void {
+    const target = ev.target! as EditorTarget;
+    if (target.value === "") {
+      return;
+    }
+    const newEntries = this.value!.concat(target.value as string);
+    target.value = "";
+    this.value = newEntries;
+    fireEvent(this, "value-changed");
+    (ev.target! as LitElement).blur();
+  }
+
+  private _valueChanged(ev: Event): void {
+    const target = ev.target! as EditorTarget;
+    const newEntries = this.value!.concat();
+
+    if (target.value === "") {
+      newEntries.splice(target.index!, 1);
+    } else {
+      newEntries[target.index!] = target.value!;
```

Comment:
We should never change the input value, instead create a new value.

Reply:
Example https://github.com/home-assistant/home-assistant-polymer/blob/dev/src/components/entity/ha-entities-picker.ts#L90-L97

Reply:
OK, I see. For that to work, the `_valueChanged` method in the map card editor may require some more changes, because if `target.configValue` is defined for the element, and `target.value` is not changed, then the method immediately returns.
When I pass back the changed value through the event, it will be available as `ev.detail.value`.
I'll look into this.

Reply:
I just realised that even the standard `paper-input` components return the new value in `ev.detail.value`. Is there any particular reason why one would choose `ev.target.value` over `ev.detail.value`?

Reply:
I had to make a couple of changes to `_valueChanged` of the map card editor to make this work.

Reply:
Of course now lint was complaining that `EntitiesEditorEvent`'s `detail` doesn't have a `value`. In an attempt to make the code cleaner, I split the event handler for the entity editor from the other input elements.